`minimal-deps` CI job fails with "coroutine was never awaited"

See e.g. https://github.com/bokeh/bokeh/actions/runs/7119067114/job/19383509931?pr=13582:

```py
______________ ERROR at setup of test__request_in_session_context ______________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_runtest_hook.<locals>.<lambda> at 0x7f2d406db1f0>
when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: "Callable[[], TResult]",
        when: "Literal['collect', 'setup', 'call', 'teardown']",
        reraise: Optional[
            Union[Type[BaseException], Tuple[Type[BaseException], ...]]
        ] = None,
    ) -> "CallInfo[TResult]":
        """Call func, wrapping the result in a CallInfo.
    
        :param func:
            The function to call. Called without arguments.
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: Optional[TResult] = func()

/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/_pytest/runner.py:341: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/_pytest/runner.py:262: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/pluggy/_hooks.py:493: in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/pluggy/_manager.py:115: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/_pytest/unraisableexception.py:83: in pytest_runtest_setup
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None, None, None]:
        with catch_unraisable_exception() as cm:
            yield
            if cm.unraisable:
                if cm.unraisable.err_msg is not None:
                    err_msg = cm.unraisable.err_msg
                else:
                    err_msg = "Exception ignored in"
                msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                msg += "".join(
                    traceback.format_exception(
                        cm.unraisable.exc_type,
                        cm.unraisable.exc_value,
                        cm.unraisable.exc_traceback,
                    )
                )
>               warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object BokehTornado._cleanup_sessions at 0x7f2d4b472a40>
E               
E               Traceback (most recent call last):
E                 File "/usr/share/miniconda3/envs/bk-test/lib/python3.9/warnings.py", line 506, in _warn_unawaited_coroutine
E                   warn(msg, category=RuntimeWarning, stacklevel=2, source=coro)
E               RuntimeWarning: coroutine 'BokehTornado._cleanup_sessions' was never awaited

/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning
------------------------------ Captured log setup ------------------------------
ERROR    asyncio:base_events.py:1753 Task was destroyed but it is pending!
task: <Task pending name='Task-579' coro=<BokehTornado._cleanup_sessions() running at /usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/bokeh/server/tornado.py:721> cb=[IOLoop.add_future.<locals>.<lambda>() at /usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/tornado/ioloop.py:719]>
```

--- a/conda/environment-test-3.10.yml
+++ b/conda/environment-test-3.10.yml
@@ -13,7 +13,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado >=5
+  - tornado >=6.2
   - xyzservices>=2021.09.1
 
   # tests

--- a/conda/environment-test-3.11.yml
+++ b/conda/environment-test-3.11.yml
@@ -13,7 +13,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado >=5
+  - tornado >=6.2
   - xyzservices>=2021.09.1
 
   # tests

--- a/conda/environment-test-3.12.yml
+++ b/conda/environment-test-3.12.yml
@@ -13,7 +13,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado >=5
+  - tornado >=6.2
   - xyzservices>=2021.09.1
 
   # tests

--- a/conda/environment-test-3.9.yml
+++ b/conda/environment-test-3.9.yml
@@ -13,7 +13,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado >=5
+  - tornado >=6.2
   - xyzservices>=2021.09.1
 
   # tests

--- a/conda/environment-test-downstream.yml
+++ b/conda/environment-test-downstream.yml
@@ -12,7 +12,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado >=5
+  - tornado >=6.2
   - xyzservices>=2021.09.1
 
   # tests

--- a/conda/environment-test-minimal-deps.yml
+++ b/conda/environment-test-minimal-deps.yml
@@ -13,7 +13,7 @@ dependencies:
   - pillow >=4.0
   - python-dateutil >=2.1
   - pyyaml >=3.10
-  - tornado 5.1.*
+  - tornado 6.2.*
   - xyzservices>=2021.09.1
 
   # tests

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@ dependencies = [
     "pandas >=1.2",
     "pillow >=7.1.0",
     "PyYAML >=3.10",
-    "tornado >=5.1",
+    "tornado >=6.2",
     "xyzservices >=2021.09.1",
 ]
 authors = [

